contracts/DefaultStrategy.sol

```diff
@@ -0,0 +1,497 @@
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+import "./common/UsingRegistryUpgradeable.sol";
+import "./common/UUPSOwnableUpgradeable.sol";
+import "./interfaces/IAccount.sol";
+import "./interfaces/IGroupHealth.sol";
+import "./Managed.sol";
+import "./interfaces/IManager.sol";
+import "./interfaces/IAllowedStrategy.sol";
+
+contract DefaultStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Managed {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /**
+     * @notice Holds a group's address and votes.
+     * @param group The address of the group.
+     * @param votes The votes assigned to the group.
+     */
+    struct GroupWithVotes {
+        address group;
+        uint256 votes;
+    }
+
+    /**
+     * @notice Contract used for checking group health and rebalancing.
+     */
+    IGroupHealth public groupHealthContract;
+
+    /**
+     * @notice An instance of the Account contract this Manager manages.
+     */
+    IAccount internal account;
+
+    /**
+     * @notice Contract used for checking group health and rebalancing.
+     */
+    IAllowedStrategy public allowedStrategyContract;
+
+    /**
+     * @notice Used when attempting to withdraw from allowed strategy
+     * but group does not have enough Celo. Group either doesn't have enough stCelo
+     * or it is necessary to rebalance the group.
+     * @param group The group's address.
+     */
+    error GroupNotBalancedOrNotEnoughStCelo(address group);
+
+    /**
+     * @notice Used when there isn't enough CELO voting for a account's strategy
+     * to fulfill a withdrawal.
+     * @param group The group's address.
+     */
+    error CantWithdrawAccordingToStrategy(address group);
+
+    /**
+     * @notice Used when attempting to deposit when the total deposit amount
+     * would tip each active group over the voting limit as defined in
+     * Election.sol.
+     */
+    error NoVotableGroups();
+
+    /**
+     * @notice Emitted when a deprecated group is no longer being voted for and
+     * the contract forgets about it entirely.
+     * @param group The group's address.
+     */
+    event GroupRemoved(address indexed group);
+
+    /**
+     * @notice Used when attempting to activate a group that is already active.
+     * @param group The group's address.
+     */
+    error GroupAlreadyAdded(address group);
+
+    /**
+     * @notice Used when an attempt to remove a deprecated group from the
+     * EnumerableSet fails.
+     * @param group The group's address.
+     */
+    error FailedToRemoveDeprecatedGroup(address group);
+
+    /**
+     * @notice Used when a group does not meet the validator group health requirements.
+     * @param group The group's address.
+     */
+    error GroupNotEligible(address group);
+
+    /**
+     * @notice Used when attempting to activate a group when the maximum number
+     * of groups voted (as allowed by the Election contract) is already being
+     * voted for.
```

Comment:
Is this still necessary given the limit is being lifted?

Reply:
Please see comment below